### PR TITLE
freetype: merge alichraghi's freetype bindings

### PR DESCRIPTION
**Pull request by slimsag**
@alichraghi has been hard at work on freetype 2 bindings for Zig, and has kindly agreed to merge them into the Mach repo (for better discoverability, so we can use them in Mach more easily.) I am merely doing the merging here.

There are a few different freetype Zig bindings floating around out there, but probably none as far along as these from what I've heard. Hopefully with this, the community can centralize effort behind one set. These are of course available in a separate repo as usual: https://github.com/hexops/mach-freetype

## Approach

I've imported revision `4e2b158`. We don't want to commit any C sources or asset files, which may be large and/or change frequently, to keep this repo size small. However, we need the upstream freetype C sources as well as a few font files for testing purposes. And so:

* https://github.com/hexops/freetype hosts the upstream C sources, plus a few font files we need for testing.
* The idea is this `hexops/freetype` repo we will have a script which pulls the minimal subset of C sources we need in (so you need not download the entire freetype repository.) This repo is included as a submodule (the `upstream/` dir) and cloned by `zig build` automatically as with other submodules in Mach projects.

@alichraghi do you want to send a PR to add a script which downloads C sources / fonts (basically, what produces the `assets/` and `freetype/` folders) in the new `hexops/freetype` repo? I imagine similar to [the GLFW one](https://github.com/hexops/mach/blob/main/glfw/update-upstream.sh) maybe.

- [x] By selecting this checkbox, I agree to license my contributions to this project under the license(s) described in the LICENSE file, and I have the right to do so or have received permission to do so by an employer or client I am producing work for whom has this right. [alichraghi's agreement](https://user-images.githubusercontent.com/3173176/169763189-ff7029dd-3b58-43c3-8831-df107015fc8b.png)

### CODE REVIEW

File: freetype/build.zig
Comment (slimsag) [inline freetype/build.zig:9]: @alichraghi I'm hoping we can change this signature to match more the `glfw.link`, `gpu_dawn.link`, etc. pattern I am using in other libraries - giving an `Options` struct. Like this: https://github.com/hexops/mach/blob/f64595393b202e54f097c22e839d33db67528c6a/glfw/build.zig#L49

Thoughts?

Comment (alichraghi) [reply freetype/build.zig:9]: what options we are gonna have?

File: freetype/src/main.zig
Comment (slimsag) [inline freetype/src/main.zig:14]: Wow, I didn't know this existed. Nice!

**Comment by slimsag** [comment]
@alichraghi I force-pushed history just now, so you are properly attributed as the code author.

Once this PR looks good to you, then I will merge and (if you like) let people on /r/zig and Twitter know you contributed this & that it exists 😃

**Comment by iddev5** [comment]
Screenshot_2022-05-24-10-59-44-65
is this a visual bug or a mistake?

**Comment by alichraghi** [comment]
yeah i think completely ready to merge (sorry for late response).
the next example will be ready soon and maybe more interesting to showoff with it

**Comment by slimsag** [comment]
@iddev5 Gah, you're right, I used the wrong author. Dang. Another rewrite incoming!

**Review by alichraghi — APPROVED** [review]

**Pull request merged**
